app/scripts/controllers/network/network-controller.js

```diff
@@ -176,62 +195,119 @@ export default class NetworkController extends EventEmitter {
   }
 
+  /**
+   * Captures information about the currently selected network — namely,
+   * the network ID and whether the network supports EIP-1559 — and then uses
+   * the results of these requests to determine the status of the network.
+   */
   async lookupNetwork() {
-    // Prevent firing when provider is not defined.
+    const { chainId, type } = this.providerStore.getState();
+    let networkChanged = false;
+    let networkId;
+    let supportsEIP1559;
+    let networkStatus;
+
     if (!this._provider) {
       log.warn(
         'NetworkController - lookupNetwork aborted due to missing provider',
       );
       return;
     }
 
-    const { chainId } = this.providerStore.getState();
     if (!chainId) {
       log.warn(
         'NetworkController - lookupNetwork aborted due to missing chainId',
       );
-      this._setNetworkState('loading');
-      this._clearNetworkDetails();
+      this._resetNetworkId();
+      this._resetNetworkStatus();
+      this._resetNetworkDetails();
       return;
     }
 
-    // Ping the RPC endpoint so we can confirm that it works
-    const initialNetwork = this.networkStore.getState();
-    const { type } = this.providerStore.getState();
     const isInfura = INFURA_PROVIDER_TYPES.includes(type);
 
-    if (isInfura) {
-      this._checkInfuraAvailability(type);
-    } else {
-      this.emit(NETWORK_EVENTS.INFURA_IS_UNBLOCKED);
-    }
+    this.once(NETWORK_EVENTS.NETWORK_DID_CHANGE, () => {
+      networkChanged = true;
+    });
 
-    let networkVersion;
-    let networkVersionError;
     try {
-      networkVersion = await this._getNetworkId();
+      const results = await Promise.all([
+        this._getNetworkId(),
+        this._determineEIP1559Compatibility(),
+      ]);
+      networkId = results[0];
+      supportsEIP1559 = results[1];
+      networkStatus = NetworkStatus.Available;
     } catch (error) {
-      networkVersionError = error;
+      if (hasProperty(error, 'code')) {
+        let responseBody;
+        try {
+          responseBody = JSON.parse(error.message);
+        } catch {
+          // error.message must not be JSON
+        }
+
+        if (
+          isPlainObject(responseBody) &&
+          responseBody.error === INFURA_BLOCKED_KEY
+        ) {
+          networkStatus = NetworkStatus.Blocked;
+        } else if (error.code === INTERNAL_RPC_ERROR_CODE) {
+          networkStatus = NetworkStatus.Unknown;
+        } else {
+          networkStatus = NetworkStatus.Unavailable;
+        }
+      } else {
+        log.warn(
+          'NetworkController - could not determine network status',
+          error,
+        );
+        networkStatus = NetworkStatus.Unknown;
+      }
     }
-    if (initialNetwork !== this.networkStore.getState()) {
+
+    if (networkChanged) {
+      // If the network has changed, then `lookupNetwork` either has been or is
+      // in the process of being called, so we don't need to go further.
       return;
```

Comment:
Is this comment saying that the lookupNetwork call we are currently in will be superceded by another incoming call so don't bother setting state below because we're just running the checks above again for a new network?

Reply:
Yup, that's right!